lightning/src/ln/msgs.rs

```diff
@@ -1213,47 +1193,51 @@ impl_writeable!(OpenChannel, {
 	first_per_commitment_point,
 	channel_flags,
 	shutdown_scriptpubkey
-});
+}, {});
 
-impl_writeable!(RevokeAndACK, {
+impl_writeable_msg!(RevokeAndACK, {
 	channel_id,
 	per_commitment_secret,
 	next_per_commitment_point
-});
+}, {});
 
-impl_writeable!(Shutdown, {
+impl_writeable_msg!(Shutdown, {
 	channel_id,
 	scriptpubkey
-});
+}, {});
 
-impl_writeable!(UpdateFailHTLC, {
+impl_writeable_msg!(UpdateFailHTLC, {
 	channel_id,
 	htlc_id,
 	reason
-});
+}, {});
 
-impl_writeable!(UpdateFailMalformedHTLC, {
+impl_writeable_msg!(UpdateFailMalformedHTLC, {
 	channel_id,
 	htlc_id,
 	sha256_of_onion,
 	failure_code
-});
+}, {});
 
-impl_writeable!(UpdateFee, {
+impl_writeable_msg!(UpdateFee, {
 	channel_id,
 	feerate_per_kw
-});
+}, {});
 
-impl_writeable!(UpdateFulfillHTLC, {
+impl_writeable_msg!(UpdateFulfillHTLC, {
 	channel_id,
 	htlc_id,
 	payment_preimage
-});
+}, {});
 
+// Note that this is written as a part of ChannelManager objects, and thus cannot change its
+// serialization format in a non-backwards-compatible way.
```

Comment:
Should this say "cannot change its serialization format in a *backwards-compatible* way"? And below

Reply:
Well if the change is backwards-compatible then presumably its ok? I'm not actually sure *how* you'd make a backwards compatible change here, but I suppose its theoretically possible?

Reply:
I see. I'm just taking issue with the double negative ("cannot change in a non-backwards-compatible way") -- maybe rephrase to "can only be changed in a backwards-compatible way"